platform/packages/platform/src/reply.rs

```diff
@@ -16,36 +21,43 @@ pub fn from_instantiate<T>(api: &dyn Api, reply: Reply) -> StdResult<Instantiate
 where
     T: DeserializeOwned,
 {
-    #[derive(Message)]
-    struct ReplyData {
-        #[prost(bytes, tag = "1")]
-        pub address: Vec<u8>,
-        #[prost(bytes, tag = "2")]
-        pub data: Vec<u8>,
-    }
+    let response: MsgInstantiateContractResponse = decode(reply)?;
 
-    let response = decode::<ReplyData>(reply)?;
+    Ok(InstantiateResponse {
+        address: api.addr_validate(&response.address)?,
+        data: maybe_from_json(response.data)?,
+    })
+}
+
+pub fn from_instantiate2<T>(api: &dyn Api, reply: Reply) -> StdResult<InstantiateResponse<T>>
+where
+    T: DeserializeOwned,
+{
+    let response: MsgInstantiateContract2Response = decode(reply)?;
 
     Ok(InstantiateResponse {
-        address: api
-            .addr_validate(&String::from_utf8(response.address).map_err(|_| {
-                StdError::generic_err("Address field contains invalid UTF-8 data!")
-            })?)?,
+        address: api.addr_validate(&response.address)?,
         data: maybe_from_json(response.data)?,
     })
 }
 
+pub fn from_instantiate2_raw(
+    api: &dyn Api,
+    reply: Reply,
+) -> StdResult<InstantiateResponse<Vec<u8>>> {
```

Comment:
The level of abstraction of this package is in contrast with the raw bytes that escape with the response. We need only the contract address, right? We do not have to interpret the `result.data` in any way. Remember that the reply may let the submessage result [go up](https://docs.cosmwasm.com/docs/smart-contracts/contract-semantics#handling-the-reply). Would suggest skipping the propagation of the data up. Is there a type the serde deserializes from any input?

Reply:
`serde_json`'s `Value` but we can't use that.